saw-central/src/SAWCentral/Crucible/MIR/Builtins.hs

```diff
@@ -1740,6 +1803,198 @@ runCrucible opts simCtx globals execFeatures ovRetTpr ovSim = do
+                case mty of
+                  Mir.TyArray _ arraySz -> pure arraySz
+                  _ -> panic
+                         "setupArg"
+                         [ "ArrayShape with non-TyArray type:"
+                         , Text.pack $ show $ PP.pretty mty
+                         ]
+              (eltCty, eltScTp) <- typeShapeToSAWTypes eltShp
+              arraySzTerm <- scNat sc $ fromIntegral @Int @Natural arraySz
+              let cty = Cryptol.tSeq (Cryptol.tNum (toInteger @Int arraySz)) eltCty
+              scTp <- scVecType sc arraySzTerm eltScTp
+              pure (cty, scTp)
+
+            StructShape mty _ _ ->
+              unsupportedType mty
+            EnumShape mty _ _ _ _ ->
+              unsupportedType mty
+            TransparentShape mty _ ->
+              unsupportedType mty
+            RefShape mty _ _ _ ->
+              unsupportedType mty
+            SliceShape mty _ _ _ ->
+              unsupportedType mty
+            FnPtrShape mty _ _ ->
+              unsupportedType mty
+
+    let -- Panic if we encounter an unsupported type that should have been
+        -- caught earlier by typeShapeToSAWTypes.
+        impossibleType :: forall a. Term -> IO a
+        impossibleType ty =
+          panic
+            "setupArg"
+            [ "Type that should have been rejected by typeShapeToSAWTypes:"
+            , Text.pack $ show $ ppTerm PPS.defaultOpts ty
+            ]
+
+    let -- Convert a fresh SAWCore term to a MIR-related Crucible.RegValue,
+        -- binding fresh What4 constants as needed and associating them to the
+        -- corresponding SAWCore subterms.
+        termToMirRegValue ::
+          forall tp'.
+          -- | The TypeShape of the SAWCore term.
+          TypeShape tp' ->
+          -- | The corresponding SAWCore type.
+          Term ->
+          -- | The SAWCore term.
+          Term ->
+          IO (Crucible.RegValue Sym tp')
+        termToMirRegValue shp scTp t =
+          case shp of
+            UnitShape {} -> do
+              pure ()
+            TupleShape _ elems -> do
+              eltScTps <-
+                case asTupleType scTp of
+                  Just eltScTps -> pure eltScTps
+                  Nothing ->
+                    panic
+                      "setupArg"
+                      [ "TupleShape with non-tuple type:"
+                      , Text.pack $ show $ ppTerm PPS.defaultOpts scTp
+                      ]
+              let tupleSz = length elems
+              let ag0 = Mir.MirAggregate (fromIntegral tupleSz) mempty
+              foldM
+                (\ag (idx, eltScTp, AgElemShape _ _ shp') -> do
+                    let oneBasedIdx :: Int
+                        oneBasedIdx = fromIntegral @Word @Int (idx+1)
+                    t' <- scTupleSelector sc t oneBasedIdx tupleSz
+                    elt <- termToMirRegValue shp' eltScTp t'
+                    Mir.mirAggregate_setIO bak idx 1 (shapeType shp') elt ag)
```

Comment:
It would probably be worth giving `1` a meaningful name here, and adding a comment on how this doesn't account for proper layout rules, perhaps referencing https://github.com/GaloisInc/crucible/pull/1503 in the process (unless there's a more authoritative source of information on `crucible-mir`'s current approach to tuple layout).

Reply:
Fair point. Since https://github.com/GaloisInc/crucible/pull/1503 is a now-closed issue, I think I'll cite `Note [union representation]` in `crucible-mir:Mir.TransTy` instead as an authoritative source.